Make sure to only generate addresses using the Hardened index convention

KtorZ opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->


# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->
- be2cd3d3b2b921db2d674cc031ff3c9c2ace536c
  :round_pushpin: **Make sure to only generate addresses using the Hardened index convention**
  It seems that the old / legacy wallet has trouble digesting addresses that are using indexes in the
soft convention. Note that the derivation is always hardened, regardless of the index value but
the convention was switched a while ago on -sl to use only indexes beyond 2^32...

Note that there are still many addresses on the current Mainnet that do not abide by this convention
and these addresses seems to break legacy wallet. So, if we can avoid generating potentially
problematic addresses in the new version, that's good :) ...

We still recognize address and account indexes across the whole domain though

- 91ecb305093549121ebca03c30eb6b0292789dd2
  :round_pushpin: **hack, hopefully temporarily, rnd state reading from the database**
  See comment in DB/Sqlite.

- 5e7a6041610918d9df41d0d3a15940422de5b010
  :round_pushpin: **re-generate API golden test for PostRandomAddressData**
  This now contains indexes that are considered invalid :)...

- b93c623486ddc26edd3415d74c6e82b2be632162
  :round_pushpin: **remove soft derivation index addresses from known addresses**
    smoke tested with wallet made from the previous release:

  ```
  $ cardano-wallet address list aef25f56c818fc31494183702016b82227cd4702
  Ok.
  [
    {
        "state": "unused",
        "id": "2w1sdSJu3GVfRsedS1xiTkpHAg8DuBZZdEXVY971uAdebdmyD75c271fMdj2j33dndqKchWis5Leaxzkxc6ksi2uupu68Ykk7V1"
    },
    {
        "state": "unused",
        "id": "2w1sdSJu3GVix5rkaLKd4fNMbUsY9ufWz1A6zWq6nyG3QFcris1rW9yaipuHDfrL8NgBWStDmAwm7TdSpqbcsMYo333tudcQYcV"
    }
  ]

  -- Restarting the server using the new version, migration applies automatically.

  $ ./cardano-wallet address list aef25f56c818fc31494183702016b82227cd4702
  Ok.
  []
  ```


# Comments

<!-- Additional comments or screenshots to attach if any -->

```
$ cardano-wallet-byron address create 924e219963cfcb5e36e56447bf63848a21134c75
Please enter your passphrase: **********
Ok.
{
    "state": "unused",
    "id": "DdzFFzCqrhsv6mq5SVNwaUkkFXBAcVpss3tmWXhxMLRgQ17iMRWEMik3a1bDYuymnQ8cgP7N4uCa5BFFBXK2raAPzqsEVZdPNKxNU5H9"
}
```

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Assign the PR to a corresponding milestone
 ✓ Acknowledge any changes required to the Wiki
-->

KtorZ committed:
Make sure to only generate addresses using the Hardened index convention

It seems that the old / legacy wallet has trouble digesting addresses that are using indexes in the
soft convention. Note that the derivation is always hardened, regardless of the index value but
the convention was switched a while ago on -sl to use only indexes beyond 2^32...

Note that there are still many addresses on the current Mainnet that do not abide by this convention
and these addresses seems to break legacy wallet. So, if we can avoid generating potentially
problematic addresses in the new version, that's good :) ...

We still recognize address and account indexes across the whole domain though

Anviking reviewed (approved):
We might want to overwrite the previous "invalid" account index in the DB.

KtorZ committed:
hack, hopefully temporarily, rnd state reading from the database

See comment in DB/Sqlite.

KtorZ committed:
re-generate API golden test for PostRandomAddressData

This now contains indexes that are considered invalid :)...

KtorZ committed:
remove soft derivation index addresses from known addresses

  smoke tested with wallet made from the previous release:

  ```
  $ cardano-wallet address list aef25f56c818fc31494183702016b82227cd4702
  Ok.
  [
    {
        "state": "unused",
        "id": "2w1sdSJu3GVfRsedS1xiTkpHAg8DuBZZdEXVY971uAdebdmyD75c271fMdj2j33dndqKchWis5Leaxzkxc6ksi2uupu68Ykk7V1"
    },
    {
        "state": "unused",
        "id": "2w1sdSJu3GVix5rkaLKd4fNMbUsY9ufWz1A6zWq6nyG3QFcris1rW9yaipuHDfrL8NgBWStDmAwm7TdSpqbcsMYo333tudcQYcV"
    }
  ]

  -- Restarting the server using the new version, migration applies automatically.

  $ ./cardano-wallet address list aef25f56c818fc31494183702016b82227cd4702
  Ok.
  []
  ```

KtorZ committed:
fix integration tests for addresses now using hardened indexes
